benchmarks/src/test/java/zipkin2/server/ServerIntegratedBenchmark.java

```diff
@@ -151,46 +157,62 @@ void runBenchmark(@Nullable GenericContainer<?> storage) throws Exception {
+        .withExposedPorts(9000)
+        .waitingFor(Wait.forHealthcheck());
 
-    GenericContainer<?> frontend = new GenericContainer<>(parse("ghcr.io/openzipkin/brave-example:armeria"))
-      .withNetwork(Network.SHARED)
-      .withNetworkAliases("frontend")
-      .withCommand("frontend")
-      .withExposedPorts(8081)
-      .waitingFor(Wait.forHealthcheck());
+    GenericContainer<?> frontend =
+      new GenericContainer<>(parse("ghcr.io/openzipkin/brave-example:armeria"))
+        .withNetwork(Network.SHARED)
+        .withNetworkAliases("frontend")
+        .withCommand("frontend")
+        .withExposedPorts(8081)
+        .waitingFor(Wait.forHealthcheck());
     containers.add(frontend);
 
-    GenericContainer<?> prometheus = new GenericContainer<>(parse("prom/prometheus:latest"))
-      .withNetwork(Network.SHARED)
-      .withNetworkAliases("prometheus")
-      .withExposedPorts(9090)
-      .withCopyFileToContainer(
-        MountableFile.forClasspathResource("prometheus.yml"), "/etc/prometheus/prometheus.yml");
+    // Use a quay.io mirror to prevent build outages due to Docker Hub pull quotas
+    // Use same version as in docker/examples/docker-compose-prometheus.yml
+    GenericContainer<?> prometheus =
+      new GenericContainer<>(parse("quay.io/prometheus/prometheus:v2.22.0"))
+        .withNetwork(Network.SHARED)
+        .withNetworkAliases("prometheus")
+        .withExposedPorts(9090)
+        .withCopyFileToContainer(
+          MountableFile.forClasspathResource("prometheus.yml"), "/etc/prometheus/prometheus.yml");
     containers.add(prometheus);
 
-    GenericContainer<?> grafana = new GenericContainer<>(parse("grafana/grafana:latest"))
+    // Use a quay.io mirror to prevent build outages due to Docker Hub pull quotas
+    // Use same version as in docker/examples/docker-compose-prometheus.yml
+    GenericContainer<?> grafana = new GenericContainer<>(parse("quay.io/app-sre/grafana:7.3.1"))
       .withNetwork(Network.SHARED)
       .withNetworkAliases("grafana")
       .withExposedPorts(3000)
       .withEnv("GF_AUTH_ANONYMOUS_ENABLED", "true")
       .withEnv("GF_AUTH_ANONYMOUS_ORG_ROLE", "Admin");
     containers.add(grafana);
 
-    GenericContainer<?> grafanaDashboards = new GenericContainer<>(parse("appropriate/curl:latest"))
-      .withNetwork(Network.SHARED)
-      .withCommand("/create.sh")
-      .withCopyFileToContainer(
-        MountableFile.forClasspathResource("create-datasource-and-dashboard.sh"), "/create.sh");
+    // This is an arbitrary small image that has curl installed
+    // Use a quay.io mirror to prevent build outages due to Docker Hub pull quotas
+    // Use same version as in docker/examples/docker-compose-prometheus.yml
+    GenericContainer<?> grafanaDashboards =
+      new GenericContainer<>(parse("quay.io/rackspace/curl:7.70.0"))
+        .withNetwork(Network.SHARED)
+        .withWorkingDirectory("/tmp")
+        .withLogConsumer(new Slf4jLogConsumer(LOG))
```

Comment:
this is the container that I can't tell if even worked or not despite it being fine in docker-compose externally. Even it logged nothing though the logging properties here are set to info -> stdout.

Reply:
If grafana has the zipkin dashboards, it should be working ok.